lib/Macros/Sources/SwiftMacros/DebugDescriptionMacro.swift

```diff
@@ -0,0 +1,287 @@
+        var summarySegments: [String] = []
+        for segment in descriptionString.segments {
+            switch segment {
+            case let .stringSegment(segment):
+                summarySegments.append(segment.content.text)
+            case let .expressionSegment(segment):
+                guard let labeledExpr = segment.expressions.single, labeledExpr.label == nil else {
+                    // This catches `appendInterpolation` overrides.
+                    let message: ErrorMessage = "unsupported custom string interpolation expression"
+                    context.diagnose(node: segment, message: message)
+                    return []
+                }
+
+                let expr = labeledExpr.expression
+
+                var propertyChain: [String] = []
+                if let declRef = expr.as(DeclReferenceExprSyntax.self) {
+                    // A reference to a single property on self.
+                    propertyChain.append(declRef.baseName.text)
+                } else if let memberAccess = expr.as(MemberAccessExprSyntax.self) {
+                    // A chain of properties, possibly starting with an explicit self.
+                    enumerateMembers(memberAccess) { declRef in
+                        propertyChain.append(declRef.baseName.text)
+                    }
+                } else {
+                    // The expression was neither a DeclReference nor a MemberAccess.
+                    let message: ErrorMessage = "unsupported expression; stored properties only"
+                    context.diagnose(node: expr, message: message)
+                    return []
+                }
+
+                // Explicit self are removed before use by LLDB.
+                if propertyChain[0] == "self" {
+                    propertyChain.removeFirst()
+                }
+
+                // Check that the root property is not a computed property on `self`.
+                // Ideally, all properties would be checked, but a macro expansion can
+                // at best only check the properties of the type it's attached to.
+                let rootProperty = propertyChain[0]
+                guard !computedProperties.contains(rootProperty) else {
+                    let message: ErrorMessage = "cannot reference computed properties"
+                    context.diagnose(node: expr, message: message)
+                    return []
+                }
+
+                let propertyPath = propertyChain.joined(separator: ".")
+                summarySegments.append("${var.\(propertyPath)}")
+            @unknown default:
+                return []
+            }
+}
+
+        let summaryString = summarySegments.joined()
+
+        var typeIdentifier: String
+        if let typeParameters = genericParameters(declaration), typeParameters.count > 0 {
+            let typePatterns = Array(repeating: ".+", count: typeParameters.count).joined(separator: ",")
```

Comment:
Nit: if you want to you can be a bit more specific on matching the generic type names. On https://docs.swift.org/swift-book/documentation/the-swift-programming-language/summaryofthegrammar/#Generic-Parameters-and-Arguments there's documentation on what's a valid Swift identifier

Reply:
This approach follows how LLDB currently defines the patterns for builtin types (ex `("^Swift.Array<.+>$`).

Is there a case you can think of where LLDB would produce a type string (either correctly or potential bugs), where a more precise regex would prevent a false positive?